mlflow/models/model.py

```diff
@@ -737,14 +749,19 @@ def log(
                 # older tracking servers. Only print out a warning for now.
                 _logger.warning(_LOG_MODEL_METADATA_WARNING_TEMPLATE, mlflow.get_artifact_uri())
                 _logger.debug("", exc_info=True)
+
+            model_version = None
             if registered_model_name is not None:
-                mlflow.tracking._model_registry.fluent._register_model(
+                model_version = mlflow.tracking._model_registry.fluent._register_model(
                     f"runs:/{run_id}/{mlflow_model.artifact_path}",
                     registered_model_name,
                     await_registration_for=await_registration_for,
                     local_model_path=local_path,
-                )
-        return mlflow_model.get_model_info()
+                ).version
+        model_info = mlflow_model.get_model_info()
+        if model_version is not None:
+            model_info._model_version = model_version
+        return model_info
```

Comment:
This looks like like `model_version` may not be undefined. Can you adjust indent?